### DISCUSSION TITLE
Could non-js plugins enable/disable w/o reload?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6956] [Ideas]
I was happy when pure wikitext plugins no longer required reload to take effect.
That behaviour makes me wonder if/why it really is necessary for such plugins to require reload when using the Enable/Disable button in the Ctrlpanel Plugins list?

It is particularly annoying when I want to update my public/published plugins and need to do a lot of reloading to access/hide my various coding tools that the visitor should not be exposed to. It is not a one time thing because I tweak and test and realize I forgot something etc.

I.e is it possible to, just like with the *installation* of pure wikitext plugins, make enable/disable take immediate effect?

Hm, I might as well explicitly call out for some attention here because very few can answer this at all, so hope you'll excuse the shout outs: @pmario @Jermolene @saqimtiaz 

Thank you!

**Comment by Jermolene** [comment]
Hi @twMat it would take some re-engineering of things to make the plugin disabling mechanism work without a restart.

In the meantime, for your own plugins, it should be possible be able to make a button that toggles the `plugin-type` field.

**Reply by twMat** [reply]
>In the meantime, for your own plugins, it should be possible be able to make a button that toggles the plugin-type field.

Oh, as simple as that? And this is very unlikely to cause any "undesired consequences", I assume then.

**Comment by pmario** [comment]
Just a remark. I think the mechanism should be part of the core already. The theme-switcher is able to enable and disable non-js plugins. But I'm not sure if it requests a restart
